### ISSUE TITLE
test fails for urls without tld (http://localhost)

### ISSUE THREAD

**Issue by raffaelj** [glitchdigital/structured-data-testing-tool#13]
**Describe the bug**
When the input url doesn't end with a tld, it is treated as html and all tests will fail.

**To Reproduce**
* Windows, Xampp, Apache, test website is in `/htdocs/mysite`
* run apache, website is available under `http://localhost/mysite`
* data test is in `/htdocs/datatest`

```bash
cd /path/to/htdocs/datatest
npm init
npm install structured-data-testing-tool
```

* Now I tried an example snippet from `README.md`

`/datatest/index.js`:

```js
const { structuredDataTest } = require('structured-data-testing-tool');
const { Twitter, Facebook } = require('structured-data-testing-tool/presets');

const url = 'http://localhost/mysite';

structuredDataTest(url, { presets: [ Twitter, Facebook ] })
.then(res => {
  console.log("All tests passed.");
})
.catch(err => {
  if (err.type === 'VALIDATION_FAILED') {
    err.res.failed.forEach(test => {
      console.error(test);
    })
  } else {
    console.log(err);
  }
});
```

* run `node index.js` --> all tests failed

If I use the html output of my site as input string, everything works like expected.

**Solution 1**

When I write my test, I know, that I always expect a url, so I can skip that check by calling `structuredDataTestUrl` directly.

```js
const { structuredDataTestUrl } = require('structured-data-testing-tool');
const url = 'http://localhost/mysite';
structuredDataTestUrl (url, {/* ... */}).then(/* ... */).catch(/* ... */);
```

**Solution 2**

Add options to `isURL` check to ignore tld:

https://github.com/glitchdigital/structured-data-testing-tool/blob/master/index.js#L434

```js
if (validator.isURL(input, { require_tld: false )) {/* ... */}
```

And thanks for your tool. Now, that I found my issue, I can fiddle and explore some more of it's features.

**Comment by iaincollins** [comment]
Thank you @raffaelj, this was an absolutely brilliant bug report!

I really appreciate the time researching the fix and the example code to replicate the issue.

_(* Sorry about all those lines of code in one file - I wrote it and even I struggle with figuring out what is going on there when I haven't looked at it in a few weeks.)_

I was able to use the report to write a test, confirm I could replicate the issue, then look up the option suggested and confirm the proposed fix resolved the bug and just [published the fix as v3.6](https://github.com/glitchdigital/structured-data-testing-tool/releases/tag/v3.6.0).

**Comment by raffaelj** [comment]
Thanks for the flowers and for the fast fix. I'm glad I was able to help.

> Sorry about all those lines of code in one file - I wrote it and even I struggle with figuring out what is going on there when I haven't looked at it in a few weeks.

To be honest: Your code is a mess. I needed four text editor instances on two monitors to track your code down when I tried to reproduce your validation mechanisms. So I rewrote your tool and the web-auto-extractor into a more readable style. I'll open a new issue for this topic with feedbask soon.
